### PR TITLE
[ntuple] Only give const ref access to fields in RNTupleModel

### PR DESCRIPTION
**Pull request by jblomer**

### CODE REVIEW

File: tree/ntuple/v7/inc/ROOT/RNTupleModel.hxx [hunk 1]
```diff
@@ -197,17 +215,21 @@ private:
    /// Throws an RException if fDefaultEntry is nullptr
    void EnsureNotBare() const;
 
-   RNTupleModel();
+   /// The field name can be a top-level field or a nested field. Returns nullptr if the field is not in the model.
+   Detail::RFieldBase *FindField(std::string_view fieldName) const;
```
Comment (hahnjo) [inline tree/ntuple/v7/inc/ROOT/RNTupleModel.hxx:219]: What is the difference between `FindField` and `GetField`?

File: tree/ntuple/v7/inc/ROOT/RNTupleModel.hxx [hunk 2]
```diff
@@ -172,6 +173,23 @@ public:
                                       std::function<std::string(const std::string &)> mapping);
    };
 
+   /// Provides mutable access to the field tree of the model. This is necessary to commit clusters during writing
+   /// and to set the on-disk field IDs when connecting a model to a page source or sink.
+   /// Attorney-Client limited friendship pattern.
```
Comment (hahnjo) [inline tree/ntuple/v7/inc/ROOT/RNTupleModel.hxx:178]: Hm, TIL about this pattern... Would it instead be feasible to have both
```c++
const RFieldZero &GetFieldZero() const { return *fFieldZero; }
RFieldZero &GetFieldZero() { return *fFieldZero; }
```
and ensure that all places that need have a non-`const` version of the the model?

File: tree/ntuple/v7/inc/ROOT/RNTupleModel.hxx [hunk 3]
```diff
@@ -322,7 +344,9 @@ public:
    std::unique_ptr<REntry> CreateBareEntry() const;
    REntry *GetDefaultEntry() const;
 
-   RFieldZero *GetFieldZero() const { return fFieldZero.get(); }
+   Detail::RFieldBase::RBulk GenerateBulk(std::string_view fieldName);
```
Comment (hahnjo) [inline tree/ntuple/v7/inc/ROOT/RNTupleModel.hxx:347]: I don't see the immediate connection of `GenerateBulk` to changing `GetFieldZero` - is this required here or did it just happen to end up in the same commit?

File: tree/ntuple/v7/inc/ROOT/RNTupleModel.hxx (hunk 1 above)
Comment (jblomer) [reply tree/ntuple/v7/inc/ROOT/RNTupleModel.hxx:219]: The only difference is that `GetFields` returns a const ref and throws if the field does not exist, `FindField` is the internal interface with a pointer (null pointer if the field is not there). Hm, we could name them both `GetField`...

File: tree/ntuple/v7/inc/ROOT/RNTupleModel.hxx (hunk 3 above)
Comment (jblomer) [reply tree/ntuple/v7/inc/ROOT/RNTupleModel.hxx]: I think that's right, not needed here.

**Review by hahnjo — APPROVED** [review]
Looks good, the other points that we discussed this morning can be addressed in future PRs.

(I'll probably be confused in a few months from now why `RFieldProxy` and `GenerateBulk` were briefly introduced and then removed again...)

**Pull request merged**